File: Problems/Algorithms/1557. Minimum Number of Vertices to Reach All Nodes/min_vertices.go
func findSmallestSetOfVertices(n int, edges [][]int) []int {
    ins := make([]int, n)

    for _, edge := range edges {
        ins[edge[1]]++
    }

    results := make([]int, 0)
    for id, degree := range ins {
        if degree == 0 {
            results = append(results, id)
        }
    }
    return results
}

File: Problems/Algorithms/1557. Minimum Number of Vertices to Reach All Nodes/MinVertices.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        boolean[] exists = new boolean[n];
        for (List<Integer> edge : edges) {
            exists[edge.get(1)] = true;
        }

        List<Integer> ans = new ArrayList();
        for (int i = 0; i < n; i++) {
            if (!exists[i]) {
                ans.add(i);
            }
        }

        return ans;
    }
}
